src/loader.ts

```diff
@@ -110,20 +132,20 @@ export default class {
      * @returns A `BunPlugin` instance.
      */
     async toPlugin(): Promise<BunPlugin> {
-        const parentThis = this;
         return {
-            name: parentThis.name,
-            setup(build) {
-                build.onLoad({ filter: new RegExp(`\.(${parentThis._config.extension})$`) }, async args => {
-                    parentThis.config.importPath = args.path;
-                    await parentThis.preload();
+            name: this.name,
+            // Arrow function does not have a scope so it can access parent this
```

Comment:
The only reason I had to use `parentThis` because typescript auto completions weren't working with `this` inside function scope.

Reply:
https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-0.html#specifying-the-type-of-this-for-functions